### PR TITLE
packit: build SRPM in Copr

### PR DESCRIPTION
**Pull request by TomasTomecek**
TL;DR:
 * specify the list of deps needed to prepare SRPM in the .packit file
 * move leapp-el7toel8-deps.spec into other_specs subdir
 * update the make commands for the building of packages

Previously we have prepared SRPM "locally" and then uploaded it
to the COPR server for the building. However, since Jan 2023 we are
required to build SRPMs on the COPR server itself. As in this project
we have to manage 2 SPEC files for two buildings:
  - leapp-el7toel8-deps.spec
        Creating metapackage resolving issues with dependencies
        on the target system; created RPMs are archived and are used
        as one of sources for building of leapp-repository.
  - leapp-repository.spec
        The main spec file used for the entire project, which requires
        archive with the deps packages.
Currently it's not possible to tell COPR which specific spec file
should be used and if they are in the same directory, COPR fails with
the error message about multiple SPEC files. But COPR is ok having
multiple spec files in a project when they are in separate directories.
To fix that, we are moving the deps spec file into the separate directory.

Also, explicitly set the `_rpmfilename` macro. This is super important as
the COPR build servers are using Mock, which redefines the macro, so packages
are stored inside RPMS directory, instead of `RPMS/%{ARCH}`. The macro must be
defined with double '%'. Using just single %, the macro is expanded when
the specfile is loaded, but it is expected to be expanded during
the build process when particular subpackages (RPMs) are created, so
each RPM has the right name. Using the single %, all RPMs would have the
name of the SRPM - which means effectively that only one RPM per build
would be created. (hopefully the explanation is clear :))

### CODE REVIEW

**Review by pirat89 — DISMISSED** [review]

**Comment by pirat89** [comment]
@TomasTomecek  majority of 7to8 tests failed. I will check it why before the merge. for now I will dismiss my review so it's not merged before we check what happens

**Comment by fernflower** [comment]
@pirat89 Looks like something related to the change
```


[1;31m2022-03-15 08:10:58.685712 [ERROR][0m Actor: dnf_transaction_check

Message: DNF execution failed with non zero exit code.

STDOUT:

Last metadata expiration check: 0:00:56 ago on Tue Mar 15 08:09:59 2022.



STDERR:

No matches found for the following disable plugin patterns: subscription-manager

Transaction check: 



 Problem 1: package leapp-upgrade-el7toel8-deps-0.15.0-0.202203151157Z.2494202.HEAD.PR848.el7.noarch requires policycoreutils-python, but none of the providers can be installed

  - package leapp-upgrade-el7toel8-0.15.0-0.202203151157Z.2494202.HEAD.PR848.el7.noarch requires leapp-repository-dependencies = 6, but none of the providers can be installed

  - conflicting requests

  - problem with installed package leapp-upgrade-el7toel8-0.15.0-0.202203151157Z.2494202.HEAD.PR848.el7.noarch

 Problem 2: package leapp-upgrade-el7toel8-0.15.0-0.202203151157Z.2494202.HEAD.PR848.el7.noarch requires leapp-repository-dependencies = 6, but none of the providers can be installed

  - package leapp-upgrade-el7toel8-deps-0.15.0-0.202203151157Z.2494202.HEAD.PR848.el7.noarch requires libselinux-python, but none of the providers can be installed

  - package leapp-0.14.0-0.20220312113203281873.pr766.25.ga643a6f.el7.noarch requires leapp-repository, but none of the providers can be installed

  - conflicting requests

  - problem with installed package leapp-0.14.0-0.20220312113203281873.pr766.25.ga643a6f.el7.noarch

  - package libselinux-python-2.9-2.1.module+el8.1.0+3972+45ab4324.x86_64 is filtered out by modular filtering
```

**Comment by TomasTomecek** [comment]
I'll take a look tomorrow, hard for me to see what's wrong with the deps

**Comment by TomasTomecek** [comment]
I just tried to reproduce in a ubi7 container. I enabled your copr repo:
```
$ podman run --rm -ti registry.access.redhat.com/ubi7/ubi bash
[root@167c1bb9374c /]# cat /etc/yum.repos.d/o.repo
[copr:copr.fedorainfracloud.org:group_oamg:leapp]
name=Copr repo for leapp owned by @oamg
baseurl=https://download.copr.fedorainfracloud.org/results/@oamg/leapp/epel-7-$basearch/
type=rpm-md
skip_if_unavailable=True
gpgcheck=1
gpgkey=https://download.copr.fedorainfracloud.org/results/@oamg/leapp/pubkey.gpg
repo_gpgcheck=0
enabled=1
enabled_metadata=1
```

I am getting this error:
```
# yum install leapp\*pr848\*
...
--> Finished Dependency Resolution
Error: Package: leapp-upgrade-el7toel8-deps-0.15.0-0.20220315115728329643.pr848.112.g2494202.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
           Requires: dnf >= 4
Error: Package: leapp-upgrade-el7toel8-deps-0.15.0-0.20220315115728329643.pr848.112.g2494202.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
           Requires: python-pyudev
```

There is no dnf 4 on el7, right? Am I missing something - this also seems like a different problem than we can see above.

Will rebase this PR to see if the build error changes too.

**Comment by pirat89** [comment]
> There is no dnf 4 on el7, right? Am I missing something - this also seems like a different problem than we can see above.

DNF is in the Extras repository.

**Comment by pirat89** [comment]
We hit a bug in copr web ui. hopefully it will be ok now. let's try it.

**Comment by fernflower** [comment]
Tests failing upon artifact installation with 
```
Stderr:
---v---v---v---v---v---
Warning: Permanently added '10.31.10.33' (ECDSA) to the list of known hosts.
Error: Package: leapp-upgrade-el7toel8-0.16.0-0.202207210832Z.ccebbfa.HEAD.PR848.el7.noarch (/leapp-upgrade-el7toel8-0.16.0-0.202207210832Z.ccebbfa.HEAD.PR848.el7.noarch)
           Requires: leapp-framework < 3
           Available: python2-leapp-0.14.0-0.202207121910Z.5c398e3.HEAD.PR781.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
               leapp-framework = 3.0
           Available: python2-leapp-0.14.0-0.202207200912Z.351cfe7.HEAD.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
               leapp-framework = 3.0
           Available: python2-leapp-0.14.0-0.20220712155517550047.pr781.21.g5c398e3.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
               leapp-framework = 3.0
           Available: python2-leapp-0.14.0-0.20220719141719568755.pr784.21.g4631bd4.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
               leapp-framework = 3.0
           Available: python2-leapp-0.14.0-1.el7_9.noarch (rhel-extras)
               leapp-framework = 2.2
           Installing: python2-leapp-0.14.0-100.20220720091400871738.master.21.g351cfe7.el7.noarch (copr:copr.fedorainfracloud.org:group_oamg:leapp)
               leapp-framework = 3.0
```

**Comment by pirat89** [comment]
@fernflower thanks for the check. it needs the rebase then.

**Comment by pirat89** [comment]
@fernflower checking it, I think we will need some other changes possibly here as well. putting on my todo list for the next week

**Comment by pirat89** [comment]
This PR has been posponed so many times that we hit the problem we need to merge it :D So I am going to rebase it and will try to figure out the rest of issues we discover. I remember there was still a problem the last time I checked it.

**Review by pirat89 — CHANGES_REQUESTED** [review]
It's still broken. The last problematic part is the need for the use of two rpmbuild macros. Without them, packit/copr crashes. With them, the result rpms are invalid (have incorrect names). I will need to sync with @TomasTomecek or someone to discuss the solution. The expected names of rpms does not follow the name of the specfile but the current solution expects that resulting rpms have the very same name. There are possible multiple solutions, e.g. define the value in the Makefile already with the expected names of packages, but to be honest, I prefer to just keep it on the rpmbuild as expected and do not use any private macros.

**Comment by pirat89** [comment]
Inside the final rpms:
```
[root@localhost ~]# rpm -qa leapp-upgrade*
leapp-upgrade-el8toel9-0.17.0-0.20230103113931850332.pr848.45.ga0daf9e.el8.noarch
leapp-upgrade-el8toel9-deps-0.17.0-0.20230103113931850332.pr848.45.ga0daf9e.el8.noarch
[root@localhost ~]# rpm -ql leapp-upgrade-el8toel9 | grep "\.rpm$"
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/files/bundled-rpms/leapp-deps-el9-5.0.9-0.202301031139Z.a0daf9e.pr_848.el9.noarch.rpm
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/files/bundled-rpms/leapp-repository-deps-el9-5.0.9-0.202301031139Z.a0daf9e.pr_848.el9.noarch.rpm
```

```
[root@localhost ~]# rpm -qa leapp-upgrade*
leapp-upgrade-el8toel9-0.17.0-0.20230103113931850332.pr848.45.ga0daf9e.el8.noarch
leapp-upgrade-el8toel9-deps-0.17.0-0.20230103113931850332.pr848.45.ga0daf9e.el8.noarch
[root@localhost ~]# rpm -ql leapp-upgrade-el8toel9 | grep "\.rpm$"
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/files/bundled-rpms/leapp-deps-el9-5.0.9-0.202301031139Z.a0daf9e.pr_848.el9.noarch.rpm
/usr/share/leapp-repository/repositories/system_upgrade/el8toel9/files/bundled-rpms/leapp-repository-deps-el9-5.0.9-0.202301031139Z.a0daf9e.pr_848.el9.noarch.rpm
```

**Comment by TomasTomecek** [comment]
LGTM from my PoV

Pavel and Peťa, thank you so much for nailing this!
